src/api/app/models/workflow_artifacts_per_step.rb

```diff
@@ -0,0 +1,21 @@
+class WorkflowArtifactsPerStep < ApplicationRecord
+  belongs_to :workflow_run, required: true
+
+  serialize :artifacts, JSON
+end
+
+# == Schema Information
+#
+# Table name: workflow_artifacts_per_steps
```

Comment:
Not quite sure why you would use that implementation detail (that you show them by step) into the table name.

Reply:
The naming is a problem here.

The model, let's call it X, represents a bunch of artifacts related to one step in a JSON format  (i.e. target_package, repositories etc.)

**WorkflowArtifacts**

The first idea was to call the method `WorkflowArtifacts` plural, because is the name that better represents what I described before. Choosing this method name implies not following rails standards and changing the **inflector**.

**WorkflowArtifact**

An alternative is just to call it `WorkflowArtifact` singular. But is not so good semantically. Because one instance of that model doesn't represent one artifact but many (related to one step).

Moreover, having something like `WorkflowArtifact#artifacts` sounds weird.

**WorkflowArtifactsPerStep**

That's why I chose `WorkflowArtifactsPerStep` which semantically is perfect, follows the standard of being singular, and `WorkflowArtifactsPerStep#artifacts` sounds better.

Please let me know if any of you have a better name for the model or for the column `X#artifacts`.

Reply:
I see, thanks for the explanation.